jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -0,0 +1,109 @@
+package ml4ir.inference.tensorflow
+
+import scala.collection.JavaConverters._
+import ml4ir.inference.tensorflow.data.{ModelFeaturesConfig, StringMapExampleBuilder, TestData}
+import org.junit.Test
+import org.junit.Assert._
+import org.tensorflow.example._
+
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+@Test
+class TensorFlowInferenceIT extends TestData {
+  val classLoader = getClass.getClassLoader
+
+  def resourceFor(path: String) = classLoader.getResource(path).getPath
+
+  /**
+   * Class holding a Example and prediction from Python
+   * @param sequence the Example context
+   * @param scores the prediction scores
+   */
+  class PredictionVector(var sequence: Map[String, String], var scores: Array[Float])
```

Comment:
if you can avoid ever using "var", do - use "val" (the scala equivalent of a final variable).  But in this case, even better: use case class:

`case class PredictionVector(sequence: Map[String, String], scores: Array[Float])`

Reply:
When doing so, the compiler complains line 103 where I'm getting the value of the sequence and scores directly:
 
`queryContext.sequence.asJava`

Reply:
what does it say?  Did you make it a *case* class, not just a "class"?

Reply:
😓 case class... that was it...
Many thanks for this. Whoa.. looks like Scala has some goodies embedded!